javascript/node-oauth-demo/index.js

```diff
@@ -1,148 +1,162 @@
+/**
+ * This is a minimal demo of how to implement OAuth with Asana using Node.js + Express.
+ * 
+ * It walks through:
+ *   - Redirecting the user to Asana's authorization URL
+ *   - Handling the OAuth callback and exchanging the code for tokens
+ *   - Using the access token to make a basic API call (GET /users/me)
+ * 
+ * Tokens are stored in cookies for simplicity in this local-only demo.
+ * In a real production app, tokens should be stored securely on the server (e.g., in a database),
+ * never exposed to client-side JavaScript (via cookies or localStorage),
+ * and refresh tokens should be kept httpOnly and encrypted if possible.
+ * 
+ * For official Asana docs, see: https://developers.asana.com/docs/oauth
+ */
+
 require("dotenv").config();
-const axios = require("axios");
 const express = require("express");
+const axios = require("axios");
 const path = require("path");
 const cors = require("cors");
 const cookieParser = require("cookie-parser");
-const { v4: uuidv4 } = require("uuid");
+const crypto = require("crypto");
 
 const app = express();
+const PORT = process.env.PORT || 3000;
```

Comment:
Do we want users to set their own port if we end up hard coding the endpoints below with port `3000`? if so we probably want to adapt the urls below (`http://localhost:3000`) to adapt to the port `http://localhost:${PORT}`